File: LeetCode/0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {
        if (n == Integer.MIN_VALUE) {
            x = x * x;
            n = n / 2;
        }

        if (n < 0) {
            n *= -1;
            x = 1 / x;
        }

        double res = 1;

        while (n > 0) {
            if ((n & 1) == 1) {
                res *= x;
            } 
            n = n >> 1;
            x *= x;
        }

        return res;
    }
}

File: LeetCode/0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        res = 1
        if n < 0 :
            n *= -1
            x = 1 / x
        while n > 0:
            if (n & 1) == 1:
                res *= x
            n = n >> 1
            x *= x
        return res